File: Classes/EVEOnlineAPI/EVE/EVERefTypes.m
//
//  EVERefTypes.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/20/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVERefTypes.h"


@implementation EVERefTypesItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"refTypeID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"refTypeName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	
	return scheme;
}

@end


@implementation EVERefTypes

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"refTypes":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVERefTypesItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVERefTypes.swift
//
//  EVERefTypes.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVERefTypesItem: EVEObject {
	public var refTypeID: Int = 0
	public var refTypeName: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"refTypeID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"refTypeName":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}

public class EVERefTypes: EVEResult {
	public var refTypes: [EVERefTypesItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"refTypes":EVESchemeElementType.Rowset(elementName: nil, type: EVEOwnerIDItem.self, transformer: nil)
		]
	}
}

